lib/manager/bundler/artifacts.ts

```diff
@@ -121,16 +125,55 @@ export async function updateArtifacts(
       `gem install bundler${bundlerVersion}`,
     ];
 
-    const bundlerHostRulesVariables = findAllAuthenticatable({
+    const bundlerHostRules = findAllAuthenticatable({
       hostType: 'bundler',
-    }).reduce(
+    });
+
+    const bundlerHostRulesVariables = bundlerHostRules.reduce(
       (variables, hostRule) => ({
         ...variables,
         ...buildBundleHostVariable(hostRule),
       }),
       {} as Record<string, string>
     );
 
+    // Detect hosts with a hyphen '-' in the url.
+    // Those cannot be added with environment variables but need to be added
+    // with the bundler config
+    const bundlerHostRulesAuthCommands: string[] = bundlerHostRules.reduce(
+      (authCommands: string[], hostRule) => {
+        if (hostRule.resolvedHost.includes('-')) {
+          authCommands.push(
+            `${hostRule.resolvedHost} ${getAuthenticationHeaderValue(hostRule)}`
```

Comment:
```suggestion
            `${hostRule.resolvedHost} ${getAuthenticationHeaderValue(hostRule)}`
```

store `getAuthenticationHeaderValue(hostRule)` in a local const and reuse, so we don't need to compute twice?

Reply:
Done